lib/fastlane/plugin/ddg_apple_automation/actions/asana_upload_action.rb

```diff
@@ -0,0 +1,65 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "httparty"
+require_relative "../helper/ddg_apple_automation_helper"
+
+module Fastlane
+  module Actions
+    class AsanaUploadAction < Action
+      def self.run(params)
+        task_id = params[:task_id]
+        token = params[:asana_access_token]
+        file_name = params[:file_name]
+
+        begin
+          file = File.open(file_name)
+          url = Helper::DdgAppleAutomationHelper::ASANA_API_URL + "/tasks/#{task_id}/attachments"
+          response = HTTParty.post(url,
+                                   headers: { 'Authorization' => "Bearer #{token}" },
+                                   body: { file: file })
+
+          unless response.success?
+            UI.user_error!("Failed to upload file to Asana task: (#{response.code} #{response.message})")
+          end
+        rescue Errno::ENOENT
```

Comment:
I think at least `Errno::EPERM` could also be raised here for file that the user has no permission to access, so perhaps we could just use `rescue` here:

```suggestion
        rescue
```